### PR TITLE
policy:attach supports reentrant

### PR DESCRIPTION
**Pull request by dormanze**
## Description
fix [#16897](https://github.com/minio/minio/issues/16897) in minio.

## Motivation and Context
The attach command supports reentrant and append policy.

## How to test this PR?
First execution:
```
./mc.exe admin policy attach myminio readonly --user foobar
```
image
Second execution:
```
./mc.exe admin policy attach myminio readonly --user foobar
```
image

Querying User Information:
```
./mc.exe admin user info myminio foobar
```
image

Add writeonly and readonly policy:
```
./mc.exe admin policy attach myminio readonly writeonly --user foobar
```
image

Querying User Information:
```
./mc.exe admin user info myminio foobar
```
image

detach has been tested:
```
./mc.exe admin policy detach myminio readonly --user foobar
```
image

Querying User Information:
```
./mc.exe admin user info myminio foobar
```
image

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by nodesocket** [comment]
Awesome, I believe this fixes https://github.com/minio/minio/issues/16897. Can we get this merged and built ASAP?

**Comment by dormanze** [comment]
> Awesome, I believe this fixes [minio/minio#16897](https://github.com/minio/minio/issues/16897). Can we get this merged and built ASAP?

@harshavardhana @vadmeste @klauspost please review my pr.

File: cmd/admin-policy-attach.go [hunk 1]
```diff
@@ -70,6 +72,59 @@ func mainAdminPolicyAttach(ctx *cli.Context) error {
...
+	} else {
+		userInfo, e := client.GetUserInfo(globalContext, user)
+		fatalIf(probe.NewError(e).Trace(args...), "Unable to get user policy info")
+		existingPolicies = userInfo.PolicyName
+	}
+	updatedPolicies, e := updateCannedPolicies(existingPolicies, args.Tail())
+	if e != nil {
+		fatalIf(probe.NewError(e).Trace(args...), "Unable to update the policy")
+	}
+	// new policys already has setting, skip
```
Comment (klauspost) [inline cmd/admin-policy-attach.go:120]:
```suggestion
	// no new policies were found, skip.
```

File: cmd/admin-policy-attach.go [hunk 2]
```diff
@@ -70,6 +72,59 @@ func mainAdminPolicyAttach(ctx *cli.Context) error {
...
 
+// updateCannedPolicies is update the policy list.
+func updateCannedPolicies(existingPolicies string, policiesToAdd []string) ([]string, error) {
+	var updatedPolicies []string
+	if len(policiesToAdd) == 0 {
+		return updatedPolicies, errors.New("empty policy name is unsupported")
+	}
+	for _, p1 := range policiesToAdd {
+		found := false
+		p1 = strings.TrimSpace(p1)
```
[the remark this answers is not in the document]
Comment (dormanze) [reply cmd/admin-policy-attach.go:84]: It's an good idea.
Thank you for your advice.

**Comment by harshavardhana** [comment]
@dormanze thanks for the PR however this is not what we want with the `policy attach` command, I already fixed the helm chart to ignore the failure. `policy attach` works for new policies automatically however it rejects if same policy is applied twice.

This is as per what we have agreed till now, will need to internally discuss if the changes you proposed are appropriate and this way take some time.

Thanks for waiting.

**Comment by dormanze** [comment]
> @dormanze thanks for the PR however this is not what we want with the `policy attach` command, I already fixed the helm chart to ignore the failure. `policy attach` works for new policies automatically however it rejects if same policy is applied twice.
> 
> This is as per what we have agreed till now, will need to internally discuss if the changes you proposed are appropriate and this may take some time.
> 
> Thanks for waiting.

Thank you for your reply, I'm honored. We look forward to your discussion result.

**Comment by psztoch** [comment]
Please repair this bug. 
As you say that the command should to generate an error on the second execution, then please add switch that will allow you to execute the command successfully as the police was previously assigned.
This bug has broken various autoconfiguration scripts for people.

**Comment by klauspost** [comment]
@harshavardhana We need to decide on this.

**Pull request closed without merging**